src/runtime/HalideBuffer.h

```diff
@@ -73,13 +73,40 @@ bool any_zero(const Container &c) {
     }
     return false;
 }
+
+#ifdef HALIDE_BUFFER_DO_NOT_USE_STD_ATOMIC
```

Comment:
Should we just always do this? Either one code path or the other isn't going to get as much test coverage as if we just unconditionally used AtomicRefCount.